File: 2022/part2.js
const fs = require('fs');
const data = fs.readFileSync('input.txt', 'utf-8').split(/\n/g);

let result = [];
let subArray = [];
for (let i = 0; i < data.length; i++) {
    if (data[i] !== '') {
        subArray.push(Math.abs(data[i]));
    } else if (subArray.length > 0) {
        result.push(subArray);
        subArray = [];
    }
}

let calories = [];
for (let i = 0; i < result.length; i++) {
    calories.push(result[i].reduce((acc, i) => acc + i, 0));
}

const sortCalories = calories.sort((a, b) => b - a)

console.log("the result is: " + (sortCalories[0] + sortCalories[1] + sortCalories[2]))

File: 2022/day1/part2.py
with open('input.txt') as f: s = f.read()

data = s.splitlines()

result = []
sub_array = []
for i in data:
  if i != '':
    sub_array.append(float(i))
  elif len(sub_array) > 0:
    result.append(sub_array)
    sub_array = []

calories = []
for i in result:
  calories.append(sum(i))

sort_calories = sorted(calories, reverse=True)

answer = sort_calories[0] + sort_calories[1] + sort_calories[2]

print(answer)